### PR TITLE
fix: remove unsupported validate proto plugin

### PR DESCRIPTION
**Pull request by artemchernyak-outreach**
## What this PR does / why we need it

Remove the imports of an unsupported plugin from all generated proto files. We still need to keep the plugin because it's heavily used in Go.

We have proved out that removing this manually still keeps the clients working as expected.

## Jira ID

[DT-3283]

## Notes for your reviewers

[DT-3283]: https://outreach-io.atlassian.net/browse/DT-3283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: shell/protoc.sh [hunk 1]
```diff
@@ -211,6 +211,16 @@ if has_grpc_client "node"; then
   )
 
   "$grpc_tools_node_bin" "${ts_args[@]}" "$(get_repo_directory)$workDir$SUBDIR/"*.proto
+
+  # remove unsupported validate proto plugin
+  if pushd "$grpc_path"; then
+    for f in * **/*; do
```
Comment (jkinkead) [inline shell/protoc.sh:217]: Isn't this the same as `* */*` in bash?

Any reason not to use a `find` command, like `for f in $(find . -name '*.js')` or similar?

Comment (artemchernyak-outreach) [reply shell/protoc.sh:217]: Not quite. `* **/*` will recursively go as deep as it needs to in the directory tree. `* */*` will only go one level deep.

Generally it's considered bad taste to use `find` for things that glob can handle. Plus everything in this directory should only be auto generated stuff.

Comment (jkinkead) [reply shell/protoc.sh:217]: This isn't true with the version on macos (5.2):

```bash
bash-5.2$ mkdir -p this/is/a/long/path
bash-5.2$ touch this/is/a/long/path/example.txt
bash-5.2$ cd this/
bash-5.2$ ls **/*
long
bash-5.2$ ls * **/*
is:
a

is/a:
long
bash-5.2$ 
```

Comment (artemchernyak-outreach) [reply shell/protoc.sh]: Oh yeah, I totally forgot about `globstar` being off by default on Macs and older bash versions. This new pattern should handle all the cases.

File: shell/protoc.sh [hunk 2]
```diff
@@ -152,6 +152,10 @@ if has_feature "validation"; then
   )
 fi
 
+delete_validate() {
+  sed -i '' '/validate_pb/d' "$1" 2>/dev/null || sed -i '/validate_pb/d' "$1"
```
Comment (jkinkead) [inline shell/protoc.sh:156]: Missing argument to `-i` in second half:
```suggestion
  sed -i '' '/validate_pb/d' "$1" 2>/dev/null || sed -i '' '/validate_pb/d' "$1"
```

**Review by jkinkead — APPROVED** [review]
One bug, LGTM otherwise!

File: shell/protoc.sh (hunk 2 above)
Comment (artemchernyak-outreach) [reply shell/protoc.sh:156]: The linux version actually doesn't take a second half. That's why this is trying to do this messy `||` logic so it supports both mac and linux.

**Pull request merged**